openhands/sdk/event/llm_convertible.py

```diff
@@ -235,9 +237,20 @@ class MessageEvent(LLMConvertibleEvent):
     )
 
     @computed_field
+    @property
     def reasoning_content(self) -> str:
         return self.llm_message.reasoning_content or ""
 
+    @model_validator(mode="before")
+    @classmethod
+    def _remove_reasoning_content(cls, data):
+        # Remove the genus from input as it is generated
+        if not isinstance(data, dict):
+            return
+        data = dict(data)
+        data.pop("reasoning_content", None)
+        return data
+
     @property
```

Comment:
This does feels unrelated to the change in this PR

```suggestion
```

Reply:
Could it because of the `@property` you added to `reasoning_content`?

Reply:
Fixed in https://github.com/All-Hands-AI/agent-sdk/pull/331/commits/fc9b834b750b99320381689ef4b862a738539e2f